lang/test/org/partiql/lang/syntax/PartiQLParserSpyTest.kt

```diff
@@ -0,0 +1,183 @@
+        assertEquals(ast00, ast01)
+    }
+
+    /**
+     * This test shows that SLL and LL parsing can happen sequentially
+     */
+    @Test
+    fun manuallyShowDoubleParsingSameQueriesWorks() {
+        val query00 = "SELECT a1 FROM t1"
+        val ast00 = PartiQLVisitor(ion).visit(parser.parseUsingSLL(query00))
+        val ast01 = PartiQLVisitor(ion).visit(parser.parseUsingLL(query00))
+        assertEquals(ast00, ast01)
+    }
+
+    /**
+     * This test shows that SLL and LL parsing can happen sequentially using different queries
+     */
+    @org.junit.Test(expected = org.junit.Test.None::class)
+    fun manuallyShowDoubleParsingWorks() {
+        val query00 = "SELECT a1 FROM t1"
+        val query01 = "SELECT a2 FROM t2"
+        PartiQLVisitor(ion).visit(parser.parseUsingSLL(query00))
+        PartiQLVisitor(ion).visit(parser.parseUsingLL(query01))
+    }
+
+    /**
+     * Shows that for simple queries, we'll only parse using the SLL parser.
+     */
+    @Test
+    fun sllParsesCorrectly() {
+        // Arrange
+        val query00 = "SELECT a FROM t"
+
+        // Act
+        val ast = parser.parseAstStatement(query00)
+
+        // Assert
+        verify(exactly = 1) { parser.parseUsingSLL(query00) }
+        verify(exactly = 0) { parser.parseUsingLL(query00) }
+        assertEquals(
+            ast,
+            PartiqlAst.build {
+                query(
+                    select(
+                        project = projectList(
+                            projectExpr(
+                                id("a", caseInsensitive(), unqualified())
+                            )
+                        ),
+                        from = scan(
+                            id("t", caseInsensitive(), unqualified())
+                        )
+                    )
+                )
+            }
+        )
+    }
+
+    /**
+     * Shows we can parse using LL after SLL throws a ParseCancellationException
+     */
+    @Test
+    fun sllThrowsException() {
+        // Arrange
+        val query00 = "SELECT a FROM t"
+        every {
+            parser.parseUsingSLL(query00)
+        } throws ParseCancellationException()
+
+        // Act
+        val ast = parser.parseAstStatement(query00)
+
+        // Assert
+        verify(exactly = 1) { parser.parseUsingSLL(query00) }
+        verify(exactly = 1) { parser.parseUsingLL(query00) }
+        assertEquals(
+            ast,
+            PartiqlAst.build {
+                query(
+                    select(
+                        project = projectList(
+                            projectExpr(
+                                id("a", caseInsensitive(), unqualified())
+                            )
+                        ),
+                        from = scan(
+                            id("t", caseInsensitive(), unqualified())
+                        )
+                    )
+                )
+            }
+        )
+    }
+
+    /**
```

Comment:
Are there any queries or tests you've found that are syntactically correct (i.e. don't throw any syntax error when parsed with the LL parser) that result in an error when run with the SLL parser (likely due to some ambiguity error)?

Reply:
I have not -- however, we wouldn't want to add them to the SpyTest -- as our unit tests would rely on the internal functionality of ANTLR.

Reply:
Got it. May be helpful as a future step/TODO for us to know what sorts of syntactically correct queries require falling back to the LL parser with the current grammar.

Could cut a separate issue to check.